### PR TITLE
Validate plugin option type 'dict' / 'dictionary'

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Until now, the type `dict`/`dictionary` was not validated in `ensure_type()` in lib/ansible/config/manager.py, even though that type is used also in plugins included with ansible/ansible (see for example https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/url.py#L39).

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/config/manager.py

### CODE REVIEW

**Review by bcoca — APPROVED** [review]

File: lib/ansible/plugins/doc_fragments/shell_common.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ class ModuleDocFragment(object):
     vars:
       - name: ansible_async_dir
   environment:
-    type: dict
+    type: dict or list
```
Comment (bcoca) [inline lib/ansible/plugins/doc_fragments/shell_common.py]: these are for modules, handled by argspec, this change should not be part of this PR

Comment (felixfontein) [reply lib/ansible/plugins/doc_fragments/shell_common.py]: They are (also) used by the shell plugins (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/shell/sh.py#L15, https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/shell/powershell.py#L16, https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/shell/cmd.py#L14), and these will stop working without this change. Without this change (and the one in `shell_windows.py`) CI looks like this: https://app.shippable.com/github/ansible/ansible/runs/174226/summary/console

Comment (felixfontein) [reply lib/ansible/plugins/doc_fragments/shell_common.py]: (Also I'm not aware of any module using these docs fragments.)

Comment (s-hertel) [reply lib/ansible/plugins/doc_fragments/shell_common.py]: `type: dict or list` skips validation because that isn't a known type. The convention we've used for that in the past is `type: raw`. But environment should just be a list, since it's coming from the field attribute anyway.

Comment (felixfontein) [reply lib/ansible/plugins/doc_fragments/shell_common.py]: I changed it to `type: list` as discussed in the meeting.

**Review by s-hertel — APPROVED** [review]

**Comment by felixfontein** [comment]
@bcoca @s-hertel thanks a lot for discussing and reviewing and merging this!
